azure/biganimal-csp-setup

```diff
@@ -0,0 +1,278 @@
+#!/bin/bash
+#
+# Copyright 2021 EnterpriseDB Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to create a SPN(Service Principal Name) with enough
+# permissions in your Azure subscription for handling the BigAnimal managed
+# service, or update SPN with MS Graph API permissions.
+#
+# What it does:
+#   - assume you have already login to your Azure AD(Active Directory) directory by Azure CLI
+#   - set your Azure CLI context to the given subscription
+#   - create a new client app or update client app in the Azure AD directory
+#   - assign this client app the role of "ower" of the given subscription
+#   - grant MS Graph API permissions
+#
+# it finally outputs the:
+#   - client app Id
+#   - client app secret
+#   - your Azure subscription Id
+# These outputs will be used in the BigAnimal Signup submission form.
+#
+# For more details, please refer to
+#  https://www.enterprisedb.com/docs/edbcloud/latest/getting_started/02_connect_cloud_account
+#
+set -e
+
+display_name=""
+subscription=""
+years=""
+client_id=""
+
+spn=""
+
+CURRENT_PATH=$(pwd)
+TMPDIR=$(mktemp -d)
+function cleanup {
+  rm -rf "${TMPDIR}"
+}
+trap cleanup EXIT
+pushd "${TMPDIR}" >/dev/null || exit
+
+show_help()
+{
+  echo "Required permissions:"
+  echo "  Microsoft.Authorization/roleAssignments/write"
+  echo "Required tools:"
+  echo "  jq"
+  echo "Usage:"
+  echo "  $0 -d NAME -s SUBSCRIPTION_ID [-i CLIENT_ID] [options]"
+  echo ""
+  echo "Options:"
+  echo "  -d, --display-name: The name of Azure AD App."
+  echo "  -s, --subscription: The Azure Subscription ID used by BigAnimal."
+  echo "  -y, --years:        [Optional] The Number of years for which the credentials will be valid. Only accept positive integer value. Default: 1 year."
+  echo "  -i, --id:           [Optional] The Application (client) ID of Service Principal will be updated."
+  echo "  -h, --help:         Show this help."
+  echo ""
+}
+
+check()
```

Comment:
like `aws/biganimal-csp-setup`, can we output some messages which tell users the `setup` script is running?

Reply:
Sure! I'll add this part too!

Reply:
Added here https://github.com/EnterpriseDB/cloud-utilities/pull/49/files#diff-4a15b2ca5ecfc13050cbfef0857fc00659a5143e708f6a540bef7245de8a50f3R76-R90